src/Dispatcher/MarkBased.php

```diff
@@ -10,24 +10,24 @@ class MarkBased extends RegexBasedAbstract
     /**
      * {@inheritDoc}
      */
-    protected function dispatchVariableRoute(array $routeData, string $uri): array
+    protected function dispatchVariableRoute(array $routeData, string $uri): Result
     {
         foreach ($routeData as $data) {
             if (! preg_match($data['regex'], $uri, $matches)) {
                 continue;
             }
 
-            [$handler, $varNames] = $data['routeMap'][$matches['MARK']];
+            [$handler, $varNames, $route] = $data['routeMap'][$matches['MARK']];
 
             $vars = [];
             $i = 0;
             foreach ($varNames as $varName) {
                 $vars[$varName] = $matches[++$i];
             }
 
-            return [self::FOUND, $handler, $vars];
+            return Result::fromArray([self::FOUND, $handler, $vars, $route]);
         }
 
-        return [self::NOT_FOUND];
+        return Result::createNotFound();
```

Comment:
Instantiating an object here will impact performance since this method is called many times during dispatching.